lang/C.md

```diff
@@ -0,0 +1,112 @@
+
+Get the latest version of Mingw-w64 via [MSYS2](https://www.msys2.org/), which provides up-to-date native builds of GCC, Mingw-w64, and other helpful C++ tools and libraries. Then follow the instructions on the [MSYS2](https://www.msys2.org/) website to install Mingw-w64.
+
+Add the path to your Mingw-w64 `bin` folder to the Windows `PATH` environment variable by using the following steps:
+
+- In the Windows search bar, type 'settings' to open your Windows Settings.
+- Search for Edit environment variables for your account.
+- Choose the Path variable and then select Edit.
+- Select New and add the Mingw-w64 destination folder path to the system path. The exact path depends on which version of Mingw-w64 you have installed and where you installed it. If you used the settings above to install Mingw-w64, then add this to the path: C:\msys64\mingw64\bin.
+- Select OK to save the updated PATH. You will need to reopen any console windows for the new PATH location to be available
+
+
+**Check your MinGW installation**
+
+To check that your Mingw-w64 tools are correctly installed and available, open a new Command Prompt and type:
+
+- `g++ --version`
+- `gdb --version`
+
+If you don't see the expected output or g++ or gdb is not a recognized command, make sure your PATH entry matches the Mingw-w64 binary location where the compilers are located.
+
+#### Mac OS
+
+Navigate to [https://brew.sh/](https://brew.sh/) and install Homebrew package manager on your system. After installing Homebrew, you can use the command `brew install gcc` to install gcc compiler.
+
+**Writing your first hello world program in C:**
+
+Create a new folder. Inside the new folder, right click and create a new file 'helloworld.c'. Write the below code in your new file.
+
+```C
+#include<stdio.h>
+int main()
+{
+    printf("Hello World!");
+    return 0;
+}
+```
+Change the working directory to where you have your C program. You can do that by using the command ‘cd’, which changes the directory. We need to pass the name of the directory in which the program is stored.
+
+Example: `cd Desktop`
+
+The next step is to compile the program. To do this we need to use the command gcc followed by the name of the program we are going to execute. In our case, we will use 'helloworld.c'. After this, an executable file will be created in the directory that your c file exists in with a '.exe' extension.
+
+In the next step, we can run the program. This is done by simply using the command `./<executable file>` in the terminal . On giving this we will get an output. Here, our helloworld code is executed and we got output for this code.
+
+Output - `Hello World!`
+
+Congratulations! You have successfully setup the C environment.
+
+If you’re looking for some projects in C for contribution, checkout the below projects:
+
+- [Project neovim](https://github.com/neovim/neovim)
+- [Project TDengine](https://github.com/taosdata/TDengine)
+- [Top Open Source Projects](https://github.com/trending/c)
```

Comment:
It would be nice if this one is being added. I'll let @amino19 approve before adding, though.
```suggestion
- [Top C Open Source Projects](https://github.com/trending/c)
- [C - The Algorithms](https://github.com/TheAlgorithms/C)
```